src/App.tsx

```diff
@@ -154,12 +155,43 @@ const App: React.FC = () => {
     //Checking for flexible update in play-store
     Util.startFlexibleUpdate();
 
-    //Checking for Notification permissions
-    Util.checkNotificationPermissions();
-
     //Listen to network change
     Util.listenToNetwork();
 
+    Util.checkNotificationPermissionsAndType(async (type, rewardProfileUid) => {
```

Comment:
instead of rewardProfileUid accept object params so that we can we this for other notifications